Classes/Helper/InlineHelper.php

```diff
@@ -176,31 +186,82 @@ public function getInlineElements($data, $name, $cType, $parentid = "parentid",
         // fetching the inline elements
         if ($childTable == "tt_content") {
             $sql = $GLOBALS["TYPO3_DB"]->exec_SELECTquery(
-                "*", $childTable, $parentid . " = '" . $parentUid .
+                "*",
+                $childTable,
+                $parentid . " = '" . $parentUid .
                 "' AND sys_language_uid IN (-1," . $sysLangUid . ")"
                 . ' AND ('
                 . $childTable . '.t3ver_wsid=0 OR '
                 . $childTable . '.t3ver_wsid=' . (int)$GLOBALS['BE_USER']->workspace
                 . ' AND ' . $childTable . '.pid<>-1'
                 . ')'
-                . $enableFields, "", "sorting"
+                . $enableFields,
+                "",
+                "sorting"
             );
         } else {
-            $sql = $GLOBALS["TYPO3_DB"]->exec_SELECTquery(
-                "*", $childTable, $parentid . " = '" . $parentUid .
-                "' AND parenttable = '" . $parenttable .
-                "' AND sys_language_uid IN (-1," . $sysLangUid . ")"
-                . ' AND ('
-                . $childTable . '.t3ver_wsid=0 OR '
-                . $childTable . '.t3ver_wsid=' . (int)$GLOBALS['BE_USER']->workspace
-                . ' AND ' . $childTable . '.pid<>-1'
-                . ')'
-                . $enableFields, "", "sorting"
-            );
+            $sql = $this->getSQL($parentid, $parenttable, $childTable, $parentUid, $sysLangUid, $enableFields);
         }
 
-        // and recursively add them to an array
-        $elements = array();
+
+        $elements = $this->fetchElements($name, $cType, $childTable, $sql);
+
+        if (empty($elements) && $GLOBALS['TSFE']->sys_language_mode == 'content_fallback') {
+
+            $sql = $this->getSQL($parentid,
+                $parenttable,
+                $childTable,
+                $parentUid,
+                $GLOBALS['TSFE']->sys_language_content,
+                $enableFields);
+
+
+            $elements = $this->fetchElements($name, $cType, $childTable, $sql);
+        }
+
+
+        return $elements;
+    }
+
+    /**
+     * @param $parentid
+     * @param $parenttable
+     * @param $childTable
+     * @param $parentUid
+     * @param $sysLangUid
+     * @param $enableFields
+     * @return string
+     */
+    protected function getSQL($parentid, $parenttable, $childTable, $parentUid, $sysLangUid, $enableFields)
```

Comment:
please check return value and method naming,
what is returned? a query result of what?